src/targets/Logary.Targets.Zipkin/Targets_Zipkin.fs

```diff
@@ -0,0 +1,141 @@
+﻿namespace Logary.Targets
+
+// https://github.com/racker/restkin
+// https://github.com/racker/tryfer
+/// A target for Zipkin.
+module Zipkin = 
+    open System
+    open System.IO
+    open System.Net
+    open System.Runtime.CompilerServices
+    open Hopac
+    open Hopac.Infixes
+    open Logary
+    open Logary.Internals
+    open Logary.Target
+    open Logary.Formatting
+    open Zipkin
+    open Microsoft.FSharp.Reflection
+        
+    module internal Option = 
+        let orDefault a = Option.fold (fun s t -> t) a
+    
+    open Option
+    
+    /// Configuration for Zipkin target.
+    type ZipkinConf = 
+        { baseUri : Uri
+          collectorFactory : Uri -> ISpanCollector }
+        
+        /// Create a new configuration with a base uri and an optional
+        /// rest client factory.
+        /// Default port: 9411
+        /// Default host: localhost
+        static member Create(?baseUri, ?fac, ?traceReq, ?formatter) = 
+            { baseUri = defaultArg baseUri (Uri "http://localhost:9411")
+              collectorFactory = defaultArg fac (fun uri -> upcast HttpCollector uri) }
+        
+        static member Default = ZipkinConf.Create()
+    
+    type private ZipkinState = 
+        { activeSpans : Map<uint64, Span>
+          collector : ISpanCollector }
+    
+    let private makeSpan (message: Message): Span = 
+        let (Int64 traceId) = message.context.["traceId"]
```

Comment:
To be created span needs IDs of the current trace (serie of rpc requests started by some user action) and current span (which determines part of the overall distributed RPC request executed by current service). Additionally service localization (ip:port) is required.

It looks like the only place for that is message context.